pkg/services/datasources/service/datasource.go

```diff
@@ -284,13 +444,39 @@ func (s *Service) DeleteDataSource(ctx context.Context, cmd *datasources.DeleteD
 	})
 }
 
+func (s *Service) getPluginContext(ctx context.Context, orgID int64, pluginID string, ds *datasources.DataSource) backend.PluginContext {
```

Comment:
there is a service to get `PluginContext`s: https://github.com/grafana/grafana/blob/plugin-validate-settings/pkg/services/pluginsintegration/plugincontext/plugincontext.go#L62 Can you use it here? It allows to retrieve context with and without DS info so you can also use it in the `AddDatasource` scenario.